UnityProject/Assets/Scripts/Core/Editor/Items/MeleeEffectEditor.cs

```diff
@@ -1,5 +1,6 @@
 using UnityEditor;
 using Weapons;
+using UnityEngine;
```

Comment:
Uhm I wonder why did we do a custom inspector for this

Reply:
MeleeEffect has quite a few fields that clog up the inspector, and not all of them are used all the time. So I made a custom inspector to just hide fields when they aren't needed so the inspector isn't as crowded with values.

Reply:
(i.e teleport config options are hidden if the weapon cannot teleport)

Reply:
Might be before `NaughtyAttributes` was added.

Reply:
It was not- however I only learnt that we had naughty attributes on this project then- Ill make the change now